### PR TITLE
Added Arquillian unit tests

### PR DESCRIPTION
**Pull request by doobrie**
I've added some Arquillian unit tests to the "ejb in war" project and modified the pom.xml as appropriate.

I thought it would be useful to have some tests on the quickstart as I can't see a situation where they wouldn't be necessary.

Hope this is useful.

### CODE REVIEW

File: ejb-in-war/pom.xml [hunk 1]
```diff
@@ -115,6 +129,119 @@
...
+            <pluginRepository>
+               <id>jboss-public-repository</id>
+               <name>JBoss Repository</name>
+               <url>http://repository.jboss.org/nexus/content/groups/public</url>
+               <releases>
+               </releases>
+               <snapshots>
+                  <enabled>false</enabled>
+               </snapshots>
+            </pluginRepository>
```
Comment (pmuir) [inline ejb-in-war/pom.xml]: Arquillian and JBoss AS connectors are now in Maven central so you can remove the repository definition

File: ejb-in-war/src/test/java/org/jboss/as/quickstarts/ejbinwar/controller/GreeterTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2011,
+ * @author JBoss, by Red Hat.
+ */
+package org.jboss.as.quickstarts.ejbinwar.controller;
```
Comment (pmuir) [inline ejb-in-war/src/test/java/org/jboss/as/quickstarts/ejbinwar/controller/GreeterTest.java]: The convention we are using with the quickstarts is to put the tests in a "test" package e.g. `org.jboss.as.quickstarts.ejbinwar.test` and not in the same package as the app clases.

File: ejb-in-war/src/test/java/org/jboss/as/quickstarts/ejbinwar/ejb/GreeterEJBTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2011,
+ * @author JBoss, by Red Hat.
+ */
+package org.jboss.as.quickstarts.ejbinwar.ejb;
```
Comment (pmuir) [inline ejb-in-war/src/test/java/org/jboss/as/quickstarts/ejbinwar/ejb/GreeterEJBTest.java]: Fix package as above.

**Comment by pmuir** [comment]
Apart from the minor changes in code review, looks really good, and thanks for the contribution. If you can update those points, I can incorporate this :-)

I will ask that you sign the JBoss AS contributor agreement at http://cla.jboss.org as well :-)

**Comment by doobrie** [comment]
Hi Pete, I've made the changes as you suggested and committed the code.  Hope these are OK.

I've also signed the JBoss AS contributor agreement.

**Comment by sgilda** [comment]
How do I run the tests? Is it: 'mvn clean test -Parq-jbossas-remote'?

Can you add instructions to the README.md file to explain how to run the tests? Could you also add what you would expect to see in the console when it runs successfully?

Thanks!

**Comment by doobrie** [comment]
Hi, Yes, that's how to run the tests.  I'll add something along those lines to the README.

**Comment by sgilda** [comment]
Great. You can look at some of the other quickstarts that have 
Arquillian tests to see what they show for the results.

Thanks!

**Comment by doobrie** [comment]
Hi,

I've added the documentation similar to the other quickstarts.  I've also fixed a bug I found when deploying the tests to a managed JBoss.

**Comment by sgilda** [comment]
I tested this and it works great. The README instructions are much better.

Actually, the CLA agreement is no longer sufficient. You will need to  license your contributions under the CC-BY-SA 3.0 Unported license. If you agree, can you send your email address to me at sgilda@redhat.com and I'll forward the information to our legal department.

Thanks!

**Comment by doobrie** [comment]
Hi, 

Thanks.  I've sent you an email as requested.

File: ejb-in-war/pom.xml (hunk 1 above)
Comment (pmuir) [inline ejb-in-war/pom.xml]: Can you remove these repository definitions?

Comment (doobrie) [inline ejb-in-war/pom.xml]: Hi, I've removed them.

**Comment by doobrie** [comment]
Hi, Is there anything else you need me to do to these changes to make them OK?  Thanks.

**Comment by sgilda** [comment]
Hi David,  I'm waiting for a response from Legal to make sure we're all set.
Pete, do you want to do a code review on this?

**Comment by doobrie** [comment]
Thanks.  I just wanted to check you weren't waiting on anything from me :)

**Comment by sgilda** [comment]
No. It turns out everything is set. Just waiting to see if Pete needs to 
do a code review. :-)

File: ejb-in-war/README.md [hunk 4]
```diff
@@ -55,6 +55,64 @@ To undeploy from JBoss AS, run this command:
 
     mvn jboss-as:undeploy
 
+
+Running the Arquillian tests
```
Comment (pmuir) [inline ejb-in-war/README.md:59]: Sande has recently aligned all the quickstart readmes. Could you take a look at that, and bring this one itno line as well?

**Comment by pmuir** [comment]
Code looks good. README alignment is all that is needed, then can be merged :-)

**Comment by pmuir** [comment]
merged with Sande's update to the readme.

**Pull request closed without merging**